packages/arb-bridge-eth/src/bridge.ts

```diff
@@ -0,0 +1,214 @@
+
+  // get sender from preceding MessageDelivered event
+  entity.senderAliased = rawMessage.sender;
+  entity.msgData = event.params.data;
+
+  //// get destination address and eth value by parsing the data field
+
+  // data consists of dest address 20 bytes + eth value 32 bytes (created by abi.encodePacked)
+  // ethereum.decode requires full 32 byte words for decoding, so we need to add 12 bytes of 0s as prefix
+  const completeData = new Bytes(64);
+  const zeroBytesToFillPrefix = completeData.length - event.params.data.length;
+  for (let i = 0; i < completeData.length; i++) {
+    if (i < zeroBytesToFillPrefix) {
+      completeData[i] = 0;
+    } else {
+      completeData[i] = event.params.data[i - zeroBytesToFillPrefix];
+    }
+  }
+
+  // decode it and save to EthDeposit entity
+  const decodedData = ethereum.decode("(address,uint256)", completeData);
+  if (decodedData) {
+    const decodedTuple = decodedData.toTuple();
+    entity.destAddr = decodedTuple[0].toAddress();
+    entity.value = decodedTuple[1].toBigInt();
+  }
+  entity.isClassic = false;
+  entity.timestamp = event.block.number;
+  entity.transactionHash = event.transaction.hash;
+  entity.blockCreatedAt = event.block.number;
+  entity.save();
+
+  // delete the old raw message
+  store.remove("RawMessage", id);
+}
+
+function handleNitroRetryable(event: InboxMessageDeliveredEvent, rawMessage: RawMessage): void {
+  const id = bigIntToId(event.params.messageNum);
+  const retryable = RetryableTx.parseRetryable(event.params.data);
+  if (retryable) {
+    let entity = new Retryable(id);
+    // get sender from preceding MessageDelivered event
+    entity.senderAliased = rawMessage.sender;
+    entity.value = event.transaction.value;
+    entity.isEthDeposit = retryable.dataLength == BigInt.zero();
+    entity.retryableTicketID = getL2RetryableTicketId(event.params.messageNum);
+    entity.destAddr = retryable.destAddress;
+    entity.l2Calldata = retryable.data;
+    entity.timestamp = event.block.timestamp;
+    entity.transactionHash = event.transaction.hash;
+    entity.blockCreatedAt = event.block.number;
+    entity.save();
+    // we delete the old raw message since now we saved the retryable
+    store.remove("RawMessage", id);
+  } else {
+    log.error("Not able to parse tx with id {}", [id.toString()]);
+  }
+}
+
+function handleClassicRetryable(
+  event: InboxMessageDeliveredEvent,
+  rawMessage: ClassicRawMessage
+): void {
+  const id = bigIntToId(event.params.messageNum);
+  const retryable = RetryableTx.parseRetryable(event.params.data);
+  if (retryable) {
+    // if there is no calldata, this is Eth deposit
+    if (retryable.dataLength == BigInt.zero()) {
+      // we track deposits with EthDeposit entities
+      let depositId =
+        event.transaction.hash.toHexString() + "-" + event.transaction.index.toString();
+      let deposit = new EthDeposit(depositId);
+      deposit.senderAliased = rawMessage.sender;
```

Comment:
Classic deposit, [assuming from an EOA](https://github.com/OffchainLabs/arbitrum/blob/e9bce5b49607a743e410ea214cec4ada80bf7bc3/packages/arb-bridge-eth/contracts/bridge/Inbox.sol#L260), would have an unaliased sender here
```
Unaliased: 0x0000...0000
Original: 0x1111....1111
Aliased: 0x2222....2222
```

Reply:
yep, I noticed that and gave a heads up to full stack team and @dewanshparashar here - https://offchainlabs.slack.com/archives/C048LPUQ2LA/p1671094381657479. 
  
My thinking was to store data as it's emitted, and let client code handle the differences (ie. query for aliased address && isClassic==false, and for unaliased address && isClassic==true). If we decide for that approach then definitely `senderAliased` field name is misleading :) I didn't change it to avoid breaking compatibility with client code, but it shouldn't be big deal to just rename it to `sender`.  
  
Alternative is to do aliasing/unaliasing in subgraph code and always store original address? That would simplify querying. What do you guys think is better approach?

Reply:
another concern was that implementation for classic Inbox was changed multiple times, and I think it wasn't doing sender unaliasing from start, that was introduced by one of the upgrades

Reply:
Yes, a heads-up was given. Now that we are actually integrating the subgraph with UI code, the alias/original/unaliased address concept is complicating things a little bit :)
IMHO If we think of subgraphs like a black box, the client shouldn't worry about alias/unaliasing - only the original L1 addresses should work fine for getting all the required transactions. We can definitely keep `aliasedSender` as a field (as an enhancement for sure), but being able to query through `originalSender` will give the maximum value to the end users and clients querying from the subgraph.

Reply:
Sounds good, I'll update the code to always store sender's original L1 address

Reply:
sgtm, also would be nice to add comment explaining the decision

Reply:
done